### ISSUE TITLE
compiler can emit empty error message hint

### ISSUE THREAD

**Issue by cyberthirst** [vyperlang/vyper#4313]
### Version Information

* vyper Version (output of `vyper --version`): 039d36926cf19916ada19f36b7e8ee9b09c1eec5

### What's your issue about?

```vyper
@external
def foo() -> uint256:
    d: DynArray[uint256, 10] = [d]
    return d[0]
```

```
Error compiling: tests/custom/test4.vy
vyper.exceptions.UndeclaredDefinition: 'd' has not been declared.

  contract "tests/custom/test4.vy:3", function "foo", line 3:32 
       2 def foo() -> uint256:
  ---> 3     d: DynArray[uint256, 10] = [d]
  ---------------------------------------^
       4     return d[0]

  (hint: )

```
see `  (hint: )`

### How can it be fixed?

Fill this in if you know how to fix it.

**Comment by cyberthirst** [comment]
another test case:

```vyper
import lib1

uses: lib1

@deploy
def __init__():
    lib1.__interface__(self).__init__()


exports: lib1.__interface__

#lib1.vy
k: uint256

@external
def bar():
    pass

@deploy
def __init__():
    self.k = 10
```

**Comment by charles-cooper** [comment]
fixed in #4351
